docs/content/extensions.md

```diff
@@ -0,0 +1,229 @@
+cd ocis-hello
+
+yarn install
+# this actually creates the assets
+yarn build
+
+# this will compile the assets into the binary
+make generate build
+```
+
+Two options:
+1. run only the necessery services from ocis and ocis-hello independently
+2. compile ocis with the updated ocis-hello
+
+### Option 1:
+get a list of ocis services:
+
+```console
+ps ax | grep ocis
+```
+
+Try to kill `ocis hello`
+
+Remember: for now, killing a service will cause ocis to restart it. This is subject to change.
+
+In order to be able to manage the processes ourselves we need to start them independently:
+
+`bin/ocis server` starts the same services as:
+
+```
+bin/ocis micro &
+bin/ocis phoenix &
+bin/ocis hello &
+bin/ocis reva &
+```
+
+Now we can kill the `ocis hello` and use our custom built ocis-hello binary:
+
+```console
+cd ../ocis-hello
+bin/ocis-hello server
+```
+
+## Hacking phoenix (and ocis-phoenix)
+
+Following https://github.com/owncloud/phoenix we are going to build the current phoenix
+
+```
+git clone https://github.com/owncloud/phoenix.git
+cd phoenix
+
+yarn install
+yarn dist
+```
+
+We can tell ocis to use the compiled assets:
+
+Kill `ocis phoenix`, then use the compiled assets when starting phoenix.
+
+```console
+cd ../ocis
+PHOENIX_ASSET_PATH="`pwd`/../phoenix/dist" bin/ocis phoenix
+```
+
+## The ownCloud design system
+
+The owncloud design system contains a set of recommended vue components for phoenix or your own ocis extensions. Use it for a consistent look and feel.
+
+Point your browser to https://owncloud.github.io/owncloud-design-system and check the available components. Live editing the examples in the browser is supported.
+
+note: There is a bug with navigation sub items: either click a nav item twice or refresh the page
+
+## External phoenix apps
+
+This is what hello is: copy and extend!
+
+1. Phoenix is configured using the config.json which is served by the phoenix service (either `bin/ocis phoenix` or `bin/ocis-phoenix server`)
+
+2. point ocis phoenix to the web config which you extended with an external app:
+`PHOENIX_WEB_CONFIG="`pwd`/../phoenix/config.json" PHOENIX_ASSET_PATH="`pwd`/../phoenix/dist" bin/ocis phoenix`
+
+```json
+{
+  "server": "http://localhost:9140",
+  "theme": "owncloud",
+  "version": "0.1.0",
+  "openIdConnect": {
+    "metadata_url": "http://localhost:9140/.well-known/openid-configuration",
+    "authority": "http://localhost:9140",
+    "client_id": "phoenix",
+    "response_type": "code",
+    "scope": "openid profile email"
+  },
+  "apps": [],
+  "external_apps": [
+    {
+      "id": "hello",
+      "path": "http://localhost:9105/hello.js",
+      "config": {
+        "url": "http://localhost:9105"
+      }
+    },
+    {
+      "id": "myapp",
+      "path": "http://localhost:6789/superapp.js",
+      "config": {
+        "backend": "http://someserver:1234",
+        "myconfig": "is awesome"
+      }
+    }
+  ]
+}
+```
+
+## Phoenix extension points
```

Comment:
I'd really love to see this in some generic place since we're in the point in time where we are adding new ones so not to have to adjust every repo then separately. I guess it's about time I contribute into oC docs repo.

Reply:
This totally makes sense. For ocis the docs folder of every ocis-* repo is aggregated on https://owncloud.github.io/. I assume it makes more sense to add a docs folder to the phoenix repo instead of ocis-phoenix. If we have that we can list the prominent items here and reference the phoenix documentation for the up to date full list.